kedro/extras/datasets/api/api_dataset.py

```diff
@@ -91,8 +90,10 @@ def __init__(
                 or ``AuthBase``, ``HTTPBasicAuth`` instance for more complex cases.
             timeout: The wait time in seconds for a response, defaults to 1 minute.
                 https://requests.readthedocs.io/en/master/user/quickstart/#timeouts
-            json: If true it will return the API response as JSON in Python Dict format,
-                else it will return pure text.
+            attribute: The attribute of response to return. Normally it's either `text`, which
+                returns pure text,`json`, which returns JSON in Python Dict format, `content`,
+                which returns a raw content, or `` (empty string), which returns the reponse
```

Comment:
I'm not a big fan of making `''` (empty string) the attribute for getting the `response` object itself. Can you think of something else? Maybe just `"response"`.